### PR TITLE
World options

### LINKED ISSUE #16745: "Simplified nutrition" mod crashes the game when entering character creation
When I create a world that has the simplified nutrition mod activated, attempting to create a character for that world immediately crashes the game, with a generic "cataclysm-tiles.exe has stopped working" message. Version: 0.C-14177-gae51d65

### LINKED ISSUE #16827: Crash when Creating world with Perfect Reliability mod installed
#On a fresh install, when ever I enabled the Perfect Reliability mod the game crashed when  it generated the world.

### PR DESCRIPTION
**Pull request by kevingranade**
Fixes #16745 
Fixes #16827 
This is a different fix than #17093 in that it changes mod-based options to inject themselves into the pre-existing options system instead of adding a new one.

### CODE REVIEW

File: src/consumption.cpp [hunk 1]
```diff
@@ -1,17 +1,19 @@
 #include "player.h"
-#include "monster.h"
+
+#include "addiction.h"
```
Comment (kevingranade) [inline src/consumption.cpp]: The churn here is alphabetizing the includes, the only real change is adding options.h.

File: src/item_factory.cpp [hunk 2]
```diff
@@ -1,33 +1,32 @@
 #include "item_factory.h"
-#include "enums.h"
```
Comment (kevingranade) [inline src/item_factory.cpp]: Alphabetizing, adding options.h and removing game.h

**Comment by BevapDin** [comment]
If those are proper world options now, why not use the options system to display / configure them?

Granted, it increases the options listed there, but it would decrease the list of available mods. It has to appear in either place.

**Comment by kevingranade** [comment]
I'd prefer they appear in the mod interface instead of the options
interface, though I'm not insistent about that. Also I was targeting a
minimal fix.
As this is a fix for a crash bug, I'd rather land it and then have a
discussion about what menu they belong on afterwards.

**Pull request merged**